app/helpers/course_year_helper.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module CourseYearHelper
+  def get_user_lessons_and_progresses(ect_profile)
+    course_lessons = CourseLesson.where(course_module: course_modules)
+
+    user_progresses = CourseLessonProgress.where(early_career_teacher_profile: ect_profile, course_lesson: course_lessons).includes(:course_lesson)
+
+    course_lessons.map do |lesson|
+      lesson.progress = user_progresses.find { |progress| progress.course_lesson == lesson }&.progress || "not_started"
+      lesson
+    end
+  end
```

Comment:
I think this method should be commonised with the one in `course_module` - the only difference is what set of course lessons they take, and that could be an argument.